test/formula.jl

```diff
@@ -106,4 +106,9 @@ t = Terms(y ~ x1 * x2 * x3)
 ## @test t.terms == [:x2, :(x1&x2)]    # == [:(1 & x1)]
 ## @test t.eterms == [:y, :x1, :x2]
 
+@test dropterm(foo ~ 1 + bar + baz, :bar).rhs == :(1 + baz)
+@test dropterm(foo ~ 1 + bar + baz, 1).rhs == :(0 + bar + baz)
+@test_throws ArgumentError dropterm(foo ~ 0 + bar + baz, 0)
+@test_throws ArgumentError dropterm(foo ~ 0 + bar + baz, :boz)
+
```

Comment:
Should test `dropterm!` as well. Maybe ensure that the input formula is indeed modified.